### PR TITLE
Core + Extension (attempted) - Korey

### PR DESCRIPTION
**Pull request by rockettdev**

### CODE REVIEW

File: src/scrabble.js [hunk 1]
```diff
@@ -1,5 +1,137 @@
-function scrabble() {
+function isAlpha(str) {
+  return (
+    /^[A-Za-z]*$/.test(str) ||
```
Comment (desokan) [inline src/scrabble.js:3]: Nice use of regex. Can you explain what the **/^[A-Za-z]*$/.test(str)** is meant to do? Thanks.

File: src/scrabble.js [hunk 2]
```diff
@@ -1,5 +1,137 @@
...
+        }
+        count += tempCount
+      } else return 0
+    }
+  }
+  return count
+}
+
+console.log(scrabble('d{og'))
+
```
Comment (desokan) [inline src/scrabble.js:136]: Generally, I like how you used call back and higher-order functions in your program. It makes your code reusable. It can also potentially make it adaptable to other projects. Well done.

Do you plan to make the three remaining edge cases pass the test? Good luck with that.

File: src/scrabble.js [hunk 3]
```diff
@@ -1,5 +1,137 @@
...
+    str === '[' ||
+    str === ']'
+  )
+}
+
+function getLettersInBracket(fromIndex, toIndex, scrabbleArray) {
+  return scrabbleArray.slice(fromIndex, toIndex)
+}
 
+function getTotalScore(k, scrabbleArray) {
```
Comment (procrastian) [inline src/scrabble.js:15]: Was there a reason for using the parameter of k for your getTotalScore function? It can get a bit confusing later on as to whether it is the letter 'k' or the parameter k that is being passed in

File: src/scrabble.js [hunk 4]
```diff
@@ -1,5 +1,137 @@
...
+    return 5
+  }
+  if (scrabbleArray[k] === 'j' || scrabbleArray[k] === 'x') {
+    return 8
+  }
+  if (scrabbleArray[k] === 'q' || scrabbleArray[k] === 'z') {
+    return 10
+  }
+  return 0
 }
```
Comment (procrastian) [inline src/scrabble.js:61]: Nice way to organise the reference letter scores, and give a return of 0 if an invalid character in given at the same time :)

File: src/scrabble.js [hunk 5]
```diff
@@ -1,5 +1,137 @@
...
+          return score
+        }
+      }
+
+      if (isAlpha(scrabbleArray[k])) {
+        tempCount += getTotalScore(k, scrabbleArray)
+        if (
+          (scrabbleArray[k - 1] === '{' && scrabbleArray[k + 1] !== '}') ||
+          (scrabbleArray[k - 1] !== '{' && scrabbleArray[k + 1] === '}')
+        ) {
```
Comment (procrastian) [inline src/scrabble.js:112]: nice way to ensure the brackets have been applied correctly!

**Review by procrastian — COMMENTED** [review]
Nice job Korey! Was able to follow the logic through nicely, and good use of outside knowledge/research with the regex!
Using the parameter k to represent the letter you're looking for confused me a bit at the start, but I can see how it works.
Good luck finishing the extension!

File: src/scrabble.js (hunk 1 above)
Comment (rockettdev) [reply src/scrabble.js:3]: In essence, it checks from a range of A-Z characters and a-z also for case sensitive edge cases

File: src/scrabble.js (hunk 3 above)
Comment (rockettdev) [reply src/scrabble.js:15]: Yeah, that was probably something i should change as it made sense in my last iteration

File: src/scrabble.js (hunk 4 above)
Comment (rockettdev) [reply src/scrabble.js:61]: Thanks!
